`Linux::release / Validate Artifact Structure / Validate artifact` structure fails on main

`.github/workflows/test_artifacts_structure.yml` is failing with

```
Current runner version: '2.335.1'
Runner name: 'aws-linux-scale-rocm-prod-7lx62-runner-v6sqv'
Runner group name: 'therock-ci-aws-prod'
Machine name: 'aws-linux-scale-rocm-prod-7lx62-runner-v6sqv'
GITHUB_TOKEN Permissions
Secret source: Actions
Prepare workflow directory
Error: Jobs without a job container are forbidden on this runner, please add a 'container:' to your job or contact your self-hosted runner administrator.
```

https://github.com/ROCm/TheRock/actions/runs/28020119255/job/82954663215

Still failing after https://github.com/ROCm/TheRock/pull/6052, with a different error now. Commented at https://github.com/ROCm/TheRock/pull/6052#discussion_r3468760384.

https://github.com/ROCm/TheRock/actions/runs/28036338651/job/83016431474
```
Run python -m pytest tests/test_artifact_structure.py \
Run '/home/runner/k8s/index.js'
============================= test session starts ==============================
platform linux -- Python 3.12.13, pytest-9.0.3, pluggy-1.6.0 -- /usr/local/bin/python
cachedir: .pytest_cache
rootdir: /__w/TheRock/TheRock
plugins: timeout-2.4.0, cov-6.0.0, check-2.5.3
timeout: 300.0s
timeout method: signal
timeout func_only: False
collecting ... collected 2 items

tests/test_artifact_structure.py::TestArtifactStructure::test_no_cross_artifact_overlaps ERROR [ 50%]
tests/test_artifact_structure.py::TestArtifactStructure::test_no_within_artifact_component_overlaps ERROR [100%]

==================================== ERRORS ====================================
___ ERROR at setup of TestArtifactStructure.test_no_cross_artifact_overlaps ____

    @pytest.fixture(scope="session")
    def artifacts_dir() -> Path:
        if not ARTIFACTS_DIR:
            pytest.skip("THEROCK_ARTIFACTS_DIR not set")
        path = Path(ARTIFACTS_DIR).resolve()
        if not path.is_dir():
>           pytest.fail(f"THEROCK_ARTIFACTS_DIR is not a directory: {path}")
E           Failed: THEROCK_ARTIFACTS_DIR is not a directory: /__w/TheRock/TheRock/.download_cache

tests/test_artifact_structure.py:55: Failed
_ ERROR at setup of TestArtifactStructure.test_no_within_artifact_component_overlaps _

    @pytest.fixture(scope="session")
    def artifacts_dir() -> Path:
        if not ARTIFACTS_DIR:
            pytest.skip("THEROCK_ARTIFACTS_DIR not set")
        path = Path(ARTIFACTS_DIR).resolve()
        if not path.is_dir():
>           pytest.fail(f"THEROCK_ARTIFACTS_DIR is not a directory: {path}")
E           Failed: THEROCK_ARTIFACTS_DIR is not a directory: /__w/TheRock/TheRock/.download_cache

tests/test_artifact_structure.py:55: Failed
=========================== short test summary info ============================
ERROR tests/test_artifact_structure.py::TestArtifactStructure::test_no_cross_artifact_overlaps - Failed: THEROCK_ARTIFACTS_DIR is not a directory: /__w/TheRock/TheRock/.download_cache
ERROR tests/test_artifact_structure.py::TestArtifactStructure::test_no_within_artifact_component_overlaps - Failed: THEROCK_ARTIFACTS_DIR is not a directory: /__w/TheRock/TheRock/.download_cache
============================== 2 errors in 0.02s ===============================
Error: Error: failed to run script step: command terminated with non-zero exit code: error executing command [sh -e /__w/_temp/80e0f470-6f27-11f1-b41f-7bbf07d90252.sh], exit code 1
Error: Process completed with exit code 1.
Error: Executing the custom container implementation failed. Please contact your self hosted runner administrator.
```

--- a/.github/workflows/test_artifacts_structure.yml
+++ b/.github/workflows/test_artifacts_structure.yml
@@ -97,7 +97,7 @@ jobs:
             --stage=all \
             --amdgpu-families="${{ inputs.amdgpu_families }}" \
             --platform="${{ env.PLATFORM }}" \
-            --output-dir="${{ github.workspace }}" \
+            --output-dir="${GITHUB_WORKSPACE}" \
             --no-extract
           # NOTE: artifact_manager.py appends /.download_cache to --output-dir
           # in --no-extract mode, so pass workspace root here so that

